source/common/http/rest_api_fetcher.cc

```diff
@@ -30,7 +30,9 @@ void RestApiFetcher::initialize() { refresh(); }
 
 void RestApiFetcher::onSuccess(Http::MessagePtr&& response) {
   uint64_t response_code = Http::Utility::getResponseStatus(response->headers());
-  if (response_code != enumToInt(Http::Code::OK)) {
+  if (response_code == enumToInt(Http::Code::NotModified)) {
+    requestComplete();
```

Comment:
a `return;` here or it will parse the response?

Reply:
Ah, yes it should return there.